src/annotator/anchoring/pdf.ts

```diff
@@ -848,6 +855,13 @@ export async function describeShape(shape: Shape): Promise<Selector[]> {
     };
   };
 
+  const textFromRect = (textLayer: HTMLElement, rect: DOMRect) => {
```

Comment:
I noticed every time this function is called we have to create the DOMRect object from the shape object. Perhaps we could make this function accept our own type instead of DOMRect, and convert from shape to DOMRect here instead, centralizing the conditional logic here.

Reply:
The shape -> DOMRect object is different for each shape though. Where we're calling this function, we're in a switch case that knows the specific kind of shape it is working with.